Yoast/wordpress-seo
# P3-32 Create get related keyphrases basic modal dialog

## Context


* We need a basic implementation of the SEMrush modal dialog to base our work on.

**Note:** re: the styling part, this PR needs https://github.com/Yoast/javascript/pull/710

## Summary


This PR can be summarized in the following changelog entry:

* Adds a basic modal dialog for the SEMrush get related keyphrases functionality.

## Relevant technical choices:

- Uses the Gutenberg Modal component.
- Enqueues the `wp-components` stylesheet to make the Modal work also with Classic Editor.
- For now, the button and the modal dialog content are **unstyled**. Pending decision whether we should use styled-components or the CSS from the admin-redesign.
- As a consequenve, for now the modal dialog does _not_ have a min-height set.
- Also, the icon in the modal dialog header isn't visible because it needs width and height set via CSS.

## Test instructions

This PR can be tested by following these steps:

- Edit a post in Gutenberg
- Empty the focus keyphrase field
- Click the "Get related keyphrases" button below the keyphrase input field
- Observe the modal dialog does _not_ open
- Enter some focus keyphrase
- Click the button to open the modal dialog
- Observe the modal dialog appears
- Observe the temporary debug info in the content
- Check the Debug info keyphrase is correct
- Check the Debug info location is `metabox`
- Close the modal dialog
- Open the Yoast plugin sidebar
- Click the button to open the modal dialog
- Observe the modal dialog looks OK and the Debug info location is `sidebar`
- Install and activate the Classic Editor plugin
- Edit a post, insert a keyphrase 
- Click the button to open the modal dialog
- Observe the modal dialog works as expected

**To test the styling:**
- switch your JavaScript repo to the `P3-5-create-related-keyphrases-component` branch from https://github.com/Yoast/javascript/pull/710
- on Yoast SEO, run `yarn link-monorepo` and then build
- open the modal and check:
  - the modal title has the Yoast icon with the Yoast purple color
  - the modal title text is bigger, _not_ bold, and purple
  - the modal has a min-height set on its content: the value `66vh` is absolutely temporary and will be adjusted based on real needs on future iterations
- also: check the Google preview modal
- go to the Yoast sidebar and click on "Google preview"
- the Google preview modal opens
- check the modal title has now the Yoast icon 
- check the modal title text has the new styling: bigger, _not_ bold, and purple


## UI changes

## Documentation

## Quality assurance


Fixes [P3-32]
